src/Bicep.Core/TypeSystem/IResourceTypeProvider.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using Bicep.Core.Resources;
+
+namespace Bicep.Core.TypeSystem
+{
+    public interface IResourceTypeProvider
+    {
+        ResourceType LookupType(ResourceTypeReference reference);
+
+        bool HasTypeDefined(ResourceTypeReference typeReference);
```

Comment:
>HasTypeDefined [](start = 13, length = 14)

another naming nit - The word "Defined" doesn't really add anything here. Can we call it `HasType()`?